### PR TITLE
Issue22

### LINKED ISSUE #22: Updating a Wishlisted Item
**As a** product owner
**I need** to be able to update the details of a product (name, price, etc...)
**So that** the most-recent information can be viewed when the item is viewed through the wishlist
   
 ### Details and Assumptions
 * An update to the details of a product, should be triggered by the products team
   
 ### Acceptance Criteria  
   
 ```gherkin
 Given that the service is available.
 When the products team attempt to update the details of a product
 Then one of two things should happen:
 1.  If the item does not belong to any of the wishlists, then the update must fail (404_NOT_FOUND)
 2.  Else, the item details must be updated in every wishlist thereby indicating success (200_OK)

```

### PR DESCRIPTION
**Pull request by ZhiruoLi**
complete issue22:
Develop the update product function

### CODE REVIEW

File: service/models.py
Comment (sohailhodarkar) [inline service/models.py:1]: @ZhiruoLi, please get rid of lines 127 - 138. They are not required. Thanks!

**Comment by sathvikabm** [comment]
Could you please check these again
 
test_routes.py
 line 261 
 resp = self.client.post(
            **f"**{BASE_URL}/{wl.id}/products" 
           
line 271 - its wishlists/wishlist_id/products/product_id
res = self.client.put(f"{BASE_URL}/wl.id/products/{product.id}",
                              json=data, content_type="application/json")
        
line 255 
wishlist factory generates new values, can we use wl = self._create_accounts(1)[0] instead?

**Comment by sathvikabm** [comment]
routes.py
line 231 UPDATE AN ITEM -> to UPDATE A PRODUCT
all the other functions follow this standard so its better if you make it product_id too, will be easier to understand
line 235 @app.route("/wislists/int:wishlist_id/products/int:products_id", methods=["PUT"])
line 236 def update_product(wishlist_id, procuct_id):
line 249
body = request.get_json()
product.deserialize(body)
>>>> (input taken in the function product_id we need to use it )add - product.id = product_id
product.update()
we have the primary key as _id right, its essential to initial it and then update

**Comment by sathvikabm** [comment]
Could you please check these again

test_routes.py
line 261
resp = self.client.post(
f"{BASE_URL}/{wl.id}/products"

line 271 - its wishlists/wishlist_id/products/product_id
res = self.client.put(f"{BASE_URL}/wl.id/products/{product.id}",
json=data, content_type="application/json")

line 255
wishlist factory generates new values, can we use wl = self._create_accounts(1)[0] instead?

**Comment by ZhiruoLi** [comment]
> routes.py line 231 UPDATE AN ITEM -> to UPDATE A PRODUCT all the other functions follow this standard so its better if you make it product_id too, will be easier to understand line 235 @app.route("/wislists/int:wishlist_id/products/int:products_id", methods=["PUT"]) line 236 def update_product(wishlist_id, procuct_id): line 249 body = request.get_json() product.deserialize(body)
> 
> > > > > (input taken in the function product_id we need to use it )add - product.id = product_id
> > > > > product.update()
> > > > > we have the primary key as _id right, its essential to initial it and then update

Hi, thanks for reviewing.

I changed the item_id to product_id to keep consistent.
.
For line 249, I don’t think I need to add that line. I get the product instance from find(). I didn’t init a new instance. I know it is important to init id if generate a new instance by Product(), which will generate a new id also. So, to avoid mistakes, I didn’t use Product(). The product.id is equal to product_id here.

For line255 in test_routes, the _create_wishlists will also generate new values. Because the create_product()  initiates a new instance, it will generate a new id also. I have to get response to get the new id instead of using a blackbox. I have checked a lot here and made sure it worked correctly.

**Review by ZhiruoLi — COMMENTED** [review]
The last update in shows on branch instead of pull request commits. For the latest version please take a look on branch issue 22.

**Comment by sohailhodarkar** [comment]
I have resolved all merge conflicts, rectified PEP8 violations, added a "sad path" test, and made some changes (there were some errors) to the API and test cases.

As of commit **5af2802**, the code works fine and is good to merge.

Before approving the PR, and consequently merging the branch, make sure that the latest commit is **5af2802**

**Review by supreeta-anand-byatnal — APPROVED** [review]
This version works!

**Pull request merged**